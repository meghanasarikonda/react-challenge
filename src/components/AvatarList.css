body {
  text-align: center;
}

.popover {
  width: 280px;
  height: variable;
  background: rgb(44, 48, 51);
  /*drop shadow: color: rgb(102, 102, 102), size 2px, blur 10px;*/
  drop-shadow(0 0 10px 2px rgb(102, 102, 102))
  padding: 14px;
  margin-left: 40%;
  margin-top: -16px

}

.triangle {
  margin-top: -12px;
  margin-left: 49.5%;
  border-color: transparent transparent black transparent;
  border-style: solid;
  border-width: 8px 8px 8px 8px;
  height: 0px;
  width: 0px;
}


.title {
  font: 'Source Sans Pro';
  color: rgb(249, 249, 249);
  font-size: 16px;
  margin-top: 14px;
  margin-bottom: 14px;
}

.avatarlist {
  height: 60px;
  width: 60px;
  border-radius: 50%;
  border: 3px solid black;
}

.avatarlist:hover {
  border: 3px solid rgb(155, 160, 163);
  cursor: pointer;
}