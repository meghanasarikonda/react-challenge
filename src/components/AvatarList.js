import React, { Component } from 'react';
import './AvatarList.css';

class AvatarList extends Component {

  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div>
        Here comes list in a popover
      </div>
    );
  }
}

export default AvatarList;