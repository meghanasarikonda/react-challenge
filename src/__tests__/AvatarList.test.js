import React from 'react';
import ReactDOM from 'react-dom';
import AvatarList from '../components/AvatarList';
import Enzyme, { shallow, mount, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

function setup() {

  const props = {
    image: {
      src: 'https://i2.wp.com/beebom.com/wp-content/uploads/2016/01/Reverse-Image-Search-Engines-Apps-And-Its-Uses-2016.jpg',
      id: 1,
      label: 'picture'
    },
    closebox: () => {
      console.log('opened popover')
    },
    imagelist: [
      { src: 'avatar1.png', label: 'Avatar 1', id: 1 },
      { src: 'avatar2.png', label: 'Avatar 2', id: 2 },
      { src: 'avatar3.png', label: 'Avatar 3', id: 3 },
      { src: 'avatar4.png', label: 'Avatar 4', id: 4 },
      { src: 'avatar5.png', label: 'Avatar 5', id: 5 },
      { src: 'avatar6.png', label: 'Avatar 6', id: 6 }
    ],
    imageUpdater: () => {
      console.log('imageUpdater')
    },
    loadingChecker: () => {
      console.log('loadingChecker')
    }
  }
 
  const enzymeWrapper = mount(<AvatarList {...props} />)
 
  return {
    props,
    enzymeWrapper
  }
}

describe('AvatarList', () => {

  it('should render correctly', () => {
    const {enzymeWrapper} = setup();
    expect(shallowToJson(enzymeWrapper)).toMatchSnapshot();
  });

  it('should have 4 key values in state', () => {
    const {enzymeWrapper} = setup();
    expect(Object.keys(enzymeWrapper.state()).length).toBe(4);
  });

  it('should have closePopover as key in state', () => {
    const {enzymeWrapper} = setup();
    expect(enzymeWrapper.state().hasOwnProperty('closePopover')).toBe(true);
  });

  it('should have imageList as key in state', () => {
    const {enzymeWrapper} = setup();
    expect(enzymeWrapper.state().hasOwnProperty('imageList')).toBe(true);
  });

  it('should have isLoading as key in state', () => {
    const {enzymeWrapper} = setup();
    expect(enzymeWrapper.state().hasOwnProperty('isLoading')).toBe(true);
  });

  it('should have popover as key in state', () => {
    const {enzymeWrapper} = setup();
    expect(enzymeWrapper.state().hasOwnProperty('popover')).toBe(true);
  });

  it('should have handleClick method', () => {
    const {enzymeWrapper} = setup();
    expect(typeof enzymeWrapper.instance().handleClick).toBe('function');
  });

  it('should have clickoutside method', () => {
    const {enzymeWrapper} = setup();
    expect(typeof enzymeWrapper.instance().clickoutside).toBe('function');
  });

  it('should have componentWillMount method', () => {
    const {enzymeWrapper} = setup();
    expect(typeof enzymeWrapper.instance().componentWillMount).toBe('function');
  });

  it('should have componentWillUnount method', () => {
    const {enzymeWrapper} = setup();
    expect(typeof enzymeWrapper.instance().componentWillUnmount).toBe('function');
  });

  it('should have h1 tag', () => {
    const {enzymeWrapper} = setup();
    expect(enzymeWrapper.find('h1').length).toBe(1);
  });

  it('should have ul tag', () => {
    const {enzymeWrapper} = setup();
    expect(enzymeWrapper.find('ul').length).toBe(1);
  });

  it('should have li tags', () => {
    const {enzymeWrapper} = setup();
    const imageListLength = (enzymeWrapper.state().imageList).length;
    expect(enzymeWrapper.find('li').length).toBe(imageListLength);
  });

});


