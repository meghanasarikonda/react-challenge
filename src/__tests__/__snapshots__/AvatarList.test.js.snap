// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarList should render correctly 1`] = `
<AvatarList
  closebox={[Function]}
  image={
    Object {
      "id": 1,
      "label": "picture",
      "src": "https://i2.wp.com/beebom.com/wp-content/uploads/2016/01/Reverse-Image-Search-Engines-Apps-And-Its-Uses-2016.jpg",
    }
  }
  imageUpdater={[Function]}
  imagelist={
    Array [
      Object {
        "id": 1,
        "label": "Avatar 1",
        "src": "avatar1.png",
      },
      Object {
        "id": 2,
        "label": "Avatar 2",
        "src": "avatar2.png",
      },
      Object {
        "id": 3,
        "label": "Avatar 3",
        "src": "avatar3.png",
      },
      Object {
        "id": 4,
        "label": "Avatar 4",
        "src": "avatar4.png",
      },
      Object {
        "id": 5,
        "label": "Avatar 5",
        "src": "avatar5.png",
      },
      Object {
        "id": 6,
        "label": "Avatar 6",
        "src": "avatar6.png",
      },
    ]
  }
  loadingChecker={[Function]}
>
  <div>
    <div
      className="Avatarlist-triangle"
    />
    <div
      className="Avatarlist-popover"
    >
      <h1
        className="Avatarlist-title"
      >
        Choose your avatar
      </h1>
      <ul>
        <li
          className="Avatarlist-currentavatar"
          key="99"
          onClick={[Function]}
        >
          <div
            className="Avatarlist-image"
          >
            <img
              alt="Avatar 1"
              key="1"
              src="avatar1.png"
            />
          </div>
        </li>
        <li
          className="Avatarlist-avatarlist"
          key="98"
          onClick={[Function]}
        >
          <div
            className="Avatarlist-image"
          >
            <img
              alt="Avatar 2"
              key="2"
              src="avatar2.png"
            />
          </div>
        </li>
        <li
          className="Avatarlist-avatarlist"
          key="97"
          onClick={[Function]}
        >
          <div
            className="Avatarlist-image"
          >
            <img
              alt="Avatar 3"
              key="3"
              src="avatar3.png"
            />
          </div>
        </li>
        <li
          className="Avatarlist-avatarlist"
          key="96"
          onClick={[Function]}
        >
          <div
            className="Avatarlist-image"
          >
            <img
              alt="Avatar 4"
              key="4"
              src="avatar4.png"
            />
          </div>
        </li>
        <li
          className="Avatarlist-avatarlist"
          key="95"
          onClick={[Function]}
        >
          <div
            className="Avatarlist-image"
          >
            <img
              alt="Avatar 5"
              key="5"
              src="avatar5.png"
            />
          </div>
        </li>
        <li
          className="Avatarlist-avatarlist"
          key="94"
          onClick={[Function]}
        >
          <div
            className="Avatarlist-image"
          >
            <img
              alt="Avatar 6"
              key="6"
              src="avatar6.png"
            />
          </div>
        </li>
      </ul>
    </div>
  </div>
</AvatarList>
`;
